draft-ietf-quic-load-balancers.md

```diff
@@ -1198,6 +1198,87 @@ format and cryptographic protection of this information may change in future
+As such, they will route such packets consistently as long as the 4-tuple does
+not change. Note that DTLS 1.0 has been deprecated by the IETF.
+
+The first octet of every DTLS 1.0 or 1.2 datagram contains the content type.
+A QUIC-LB load balancer will interpret any content type less than 128 as a short
+header packet, meaning that the subsequent octets should contain a connection
+ID.
+
+Existing TLS content types comfortably fit in the range below 128. Assignment of
+codepoints greater than 64 would require coordination in accordance with
+[?RFC7983], and anyway would likely create problems demultiplexing DTLS and
+version 1 of QUIC. Therefore, this document believes it is extremely unlikely
+that TLS content types of 128 or greater will be assigned. Nevertheless, such
+an assignment would cause a QUIC-LB load balancer to interpret the packet as a
+QUIC long header with an essentially random connection ID, which is likely to be
+routed irregularly.
+
+The second octet of every DTLS 1.0 or 1.2 datagram is the bitwise complement
+of the DTLS Major version (i.e. version 1.x = 0xfe). A QUIC-LB load balancer
+will interpret this as a connection ID that requires 4-tuple based load
+balancing, meaning that the routing will be consistent as long as the 4-tuple
+remains the same.
+
+[?I-D.ietf-tls-dtls-connection-id] defines an extension to add connection IDs
+to DTLS 1.2. Unfortunately, a QUIC-LB load balancer will not correctly parse
+the connection ID and will continue 4-tuple routing. An modified QUIC-LB load
+balancer that correctly identifies DTLS and parses a DTLS 1.2 datagram for
+the connection ID is outside the scope of this document.
+
+## DTLS 1.3
+
+DTLS 1.3 [?I-D.draft-ietf-tls-dtls13] changes the structure of datagram headers
+in relevant ways.
+
+Handshake packets continue to have a TLS content type in the first octet and
+0xfe in the second octet, so they will be 4-tuple routed, which should not
+present problems for likely NAT rebinding or address change events.
+
+Non-handshake packets always have zero in their most significant bit and will
+therefore always be treated as QUIC short headers. If the connection ID is
+present, it follows in the succeeding octets. Therefore, a DTLS 1.3 association
+where the server utilizes Connection IDs and the encodings in this document
+will be routed correctly in the presence of client address and port changes.
+
+However, if the client does not include the connection_id extension in its
+ClientHello, the server is unable to use connection IDs. In this case, non-
+handshake packets will appear to contain random connection IDs and be routed
+randomly.
+
+Therefore, a DTLS 1.3 server operating behind a QUIC-LB load balancer, when
+processing a ClientHello with no connection_id extension, MUST either
+terminate the connection with a missing_extension alert or downgrade to DTLS
+1.2. Doing otherwise is likely to black hole the connection.
```

Comment:
NEVER recommend downgrade.  Maybe just imply that a server that requires the use of connection IDs might not be able to deploy DTLS 1.3.

Reply:
Yeah, downgrade is a gross suggestion, but it's the only way to continue. This is a case where DTLS 1.3 will often be fine but can negotiate itself into a mode that fails catastrophically.

But your second sentence is backwards. A TLS1.3 server MUST support connection IDs, or else the ciphertext packets will spray all over the place. The real problem is if the client refuses to use them.